serializers/src/main/java/io/pravega/schemaregistry/serializers/SerializerFactoryHelper.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.client.SchemaRegistryClientFactory;
+import io.pravega.schemaregistry.contract.data.CodecType;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+class SerializerFactoryHelper {
+    static SchemaRegistryClient initForSerializer(SerializerConfig config) {
+        SchemaRegistryClient schemaRegistryClient = getSchemaRegistryClient(config);
+        createGroup(schemaRegistryClient, config);
+        registerCodec(schemaRegistryClient, config);
+        return schemaRegistryClient;
+    }
+
+    static SchemaRegistryClient initForDeserializer(SerializerConfig config) {
+        SchemaRegistryClient schemaRegistryClient = getSchemaRegistryClient(config);
+
+        createGroup(schemaRegistryClient, config);
+        failOnCodecMismatch(schemaRegistryClient, config);
+        return schemaRegistryClient;
+    }
+
+    private static SchemaRegistryClient getSchemaRegistryClient(SerializerConfig config) {
+        return config.getRegistryConfigOrClient().isLeft() ?
+                SchemaRegistryClientFactory.withNamespace(config.getNamespace(), config.getRegistryConfigOrClient().getLeft()) :
+                config.getRegistryConfigOrClient().getRight();
+    }
+
+    private static void createGroup(SchemaRegistryClient client, SerializerConfig config) {
+        if (config.isCreateGroup()) {
+            client.addGroup(config.getGroupId(), config.getGroupProperties());
+        }
+    }
+
+    private static void registerCodec(SchemaRegistryClient client, SerializerConfig config) {
+        if (config.isRegisterCodec()) {
+            client.addCodecType(config.getGroupId(), config.getEncoder().getCodecType());
+        }
+    }
+
+    private static void failOnCodecMismatch(SchemaRegistryClient client, SerializerConfig config) {
+        if (config.isFailOnCodecMismatch()) {
+            List<String> codecTypesInGroup = client.getCodecTypes(config.getGroupId()).stream()
+                                                   .map(CodecType::getName).collect(Collectors.toList());
+            if (!config.getDecoders().getDecoderNames().containsAll(codecTypesInGroup)) {
+                log.warn("Not all CodecTypes are supported by reader. Required codecTypes = {}", codecTypesInGroup);
+                throw new RuntimeException(String.format("Need all codecTypes in %s", codecTypesInGroup.toString()));
```

Comment:
I'm not sure I follow how the reader can miss codecs.

Reply:
For example, if a codec called "aes encryption" is registered. And if the deserializer is not instantiated with the decoder for it, this check fails the initialization of deserializer

Reply:
> Do note that registry service registers codec types and not the actual library for encoding and decoding the data.

I get that we are not passing the lib around, but I don't understand this registration business. Why does the application need to register a codec on deserializer instantiation? Why can't it use whatever the stream events require according to the information in the registry?

Reply:
> Why does the application need to register a codec on deserializer instantiation? 

Its not registering, its merely checking that it has been provided with deserializers for all the registered codetypes. 

this is if the application wants to fail early if it doesnt have a decoder for an codec used for encoding the data.

Reply:
I think I understand it now based on the encryption example you gave elsewhere. It is possible that the application is using a custom encoder and this reader doesn't have the implementation. In this case, it probably needs to fail. Independent of the answer, I hope this is well documented somewhere, it is a bit subtle.

Reply:
Here, I need to make sure we are going to document this behavior around encoders and decoders. How do we make sure we don't miss it?

Reply:
I have updated the javadoc in Serializer config with more concrete example for the flag `failOnCodecMismatch`.

we will also include more details around this aspect in the usage documentation.